src/libs/actions/Session/index.js

```diff
@@ -23,6 +23,32 @@ Onyx.connect({
     callback: val => credentials = val || {},
 });
 
+/**
+ * Manage push notification subscriptions on sign-in/sign-out.
+ *
+ * We would normally do this when AuthScreens mounts and unmounts. However on Android,
+ * AuthScreens unmounts when the app is closed with the back button so we manage the
+ * push subscription when the session changes here.
+ */
+let previousAccountID = '';
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (session) => {
+        const accountID = lodashGet(session, 'accountID');
+
+        if (accountID && previousAccountID !== accountID) {
+            PushNotification.register(accountID);
+            previousAccountID = accountID;
+        }
+
+        if (!accountID && previousAccountID !== accountID) {
+            PushNotification.deregister();
+            PushNotification.clearNotifications();
+            previousAccountID = accountID;
```

Comment:
Is there a case where we would not want to set the `previousAccountID`? Or could we just do that once at the bottom of this function and always update it to the latest?

Reply:
I dunno if I like the "current" and "next" naming since the accountID has already changed by the time this callback runs but I do think this logic would work too